style(all): enable `eslint/curly` lint rule

camc314 opened the pull request:
built on https://github.com/oxc-project/oxc/pull/16749, we can use the linter to be consistent here

camc314 commented:
* **#16753** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16753?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16753?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=16753).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (commented):
Great! It was `oxfmt` changing max line length to 100 which made all these nasties.

overlookmotel commented:
### Merge activity

* **Dec 11, 2:48 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 11, 2:49 PM UTC**: `overlookmotel` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Dec 11, 2:51 PM UTC**: The [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) couldn't merge this PR because **it was not satisfying all requirements** (Failed CI: 'Spell Check').

camc314 committed:
style(all): enable `eslint/curly` lint rule (#16753)

built on https://github.com/oxc-project/oxc/pull/16749, we can use the linter to be consistent here
